src/main/java/roomescape/dto/response/ReservationResponse.java

```diff
@@ -0,0 +1,26 @@
+package roomescape.dto.response;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import roomescape.entity.Reservation;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+public record ReservationResponse(
+        Long id,
+
+        String name,
+
+        LocalDate date,
+
+        @JsonFormat(pattern = "HH:mm")
+        LocalTime time
+) {
+    public static ReservationResponse create(Reservation reservation) {
```

Comment:
[질문 ✋ ]
정적 팩토리 메서드를 사용하셨네요? 
- 러키가 정적 팩토리 메서드를 사용하는 기준은 무엇인가요?
- 정적 팩토리 메서드는 어떤 장점이 있나요?

Reply:
아직은 정적 팩토리 메서드를 dto <-> entity 변환에서 밖에 사용해보지 않아서 기준이 있다고 말할 수 없을 것 같아요😅
저는 dto <-> entity 변환할 때 정적 팩토리를 사용하면 캡슐화를 통해 내부 구조를 몰라도 자유롭게 형 변환이 가능하기 때문에 정적 팩토리 메서드를 사용합니다!

정적 팩토리 메서드는 이름을 가질 수 있음, 하위 타입 객체 반환 등 여러 장점이 존재하지만, 현재 로직에서 얻을 수 있는 장점은
response 생성 로직을 response 내부에 캡슐화함으로써, response의 생성 방법이 변경되더라도 서비스 계층에서의 코드 변경을 줄일 수 있습니다.

Reply:
[질문 ✋ ]

> 현재 로직에서 얻을 수 있는 장점은
response 생성 로직을 response 내부에 캡슐화함으로써, response의 생성 방법이 변경되더라도 서비스 계층에서의 코드 변경을 줄일 수 있습니다.

해당 장점은 생성자를 통해서도 얻을 수 있지 않을까요? 생성자 대신 정적 팩터리 메서드를 사용한 이유가 궁금해요! 정적 팩터리 메서드는 생성자의 의미를 간소화 시키고 생성 로직을 여러 곳으로 분산시킨다는 점에서 객체지향적 프로그래밍을 지향하신다면 초기에는 최대한 지향하고 러키만의 기준이 세워졌을 때 사용해보는 것을 추천드립니다 ㅎㅎ

```suggestion
    public ReservationResponse(Reservation reservation) {
      ...
```

Reply:
생성자로도 가능했군요..! 수정하였습니다 🫡

[질문 ✋ ]
콜리는 어떤 기준을 가지고 정적 팩터리 메서드를 사용하고 계신가요?

Reply:
[답변 🗣️ ]

정말 솔직한 마음으로는 저도 명확한 기준이 있다! 라고 할 순 없지만 정적 팩터리 메서드를 씀으로서 의도한 장점을 생성자로는 얻을 수 없을지 한번 고민해보는 것 같아요. 최대한 지양하고 상황을 기준으로 판단할때 이건 생성자가 아니라 꼭-꼭 정팩메여야 한다! 라는 판단이 들 때만 사용하게 되는 것 같습니다.

예를 들어 러키가 말한 dto to entity 나 entity to dto는 정팩메 사용에 대해 유연하게 바라보는 면도 있는 것 같아요